crates/solvers/src/util/conv.rs

```diff
@@ -0,0 +1,76 @@
+//! Conversion utilities.
+
+use crate::domain::eth;
+use bigdecimal::BigDecimal;
+use ethereum_types::U256;
+use num::{rational::Ratio, BigUint};
+
+/// Converts a `BigDecimal` value to a `eth::Rational` value. Returns `None` if
+/// the specified decimal value cannot be represented as a rational of `U256`
+/// integers.
+pub fn decimal_to_rational(d: &BigDecimal) -> Option<eth::Rational> {
```

Comment:
Are you sure you want this to fail if the value cannot be *exactly* represented? The function could round instead. 

Maybe it's fine but it might be unexpected that a value like from my previous example doesn't work. I mean a value that fits into the range of values that Rational can represent but not exactly. For example, when converting a u64 to f32 it is usually expected that the closest f32 value is chosen when it can't represent the original value exactly.

The current behavior only makes sense if it is guaranteed that the original values can be perfectly represented. Is this the case?

Reply:
> The current behavior only makes sense if it is guaranteed that the original values can be perfectly represented. Is this the case?

Yes. Rationals only appear for liquidity in ways that can be represented as U256 rationals:
- Uniswap fees (these are actually u24 rationals)
- Balancer fees and token weights (these are represented as x/1e18 where x needs to fit within a u256 because of its on-chain representation)

If we need an approximate rational, we can add that later, but for now I think this is the correct intended behaviour.